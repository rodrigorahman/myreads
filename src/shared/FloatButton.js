import React, {Component} from 'react';
import PropTypes from 'prop-types';


class FloatButton extends Component {

    static propTypes = {
        onClickFn: PropTypes.func.isRequired
    };

    render() {
        return (
            <div className="open-search">
                <a id='addBook' onClick={() => this.props.onClickFn()}>Add a book</a>
            </div>
        )
    }
}

export default FloatButton;

