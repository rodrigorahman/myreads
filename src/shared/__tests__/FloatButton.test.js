import React from 'react';

import { mount, shallow } from 'enzyme';
import FloatButton from "../FloatButton";

it('renders without crashing', () => {
    expect(shallow(<FloatButton onClickFn={() => {}}/>)).toMatchSnapshot();
});

it('Invoke click', () => {
    const onClickFn = jest.fn();
    const test = mount(<FloatButton onClick={onClickFn}/>)

    test.find('#addBook').simulate('click');

    expect(onClickFn).toHaveBeenCalledTimes(1);
});