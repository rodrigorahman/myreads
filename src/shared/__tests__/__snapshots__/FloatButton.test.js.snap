// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="open-search"
>
  <a
    id="addBook"
    onClick={[Function]}
  >
    Add a book
  </a>
</div>
`;
